tests/test_mapping.py

```diff
@@ -148,6 +148,15 @@ def test_MapSplit(self):
             self.assertEqual(split.splitMap.nOut, 1)
             self.assertEqual(split.origOut[0], i + 1)
 
+    def test_ZeroPoints(self):
+        """Test that Mapping.applyForward and applyInverse can handle zero points
+        """
+        mapping = ast.ZoomMap(2, 0.5)
+        out_points1 = mapping.applyForward([])
+        self.assertEqual(len(out_points1), 0)
+        out_points2 = mapping.applyInverse([])
+        self.assertEqual(len(out_points2), 0)
```

Comment:
Maybe test with a mapping that changes dimensions (e.g., 2 -> 1 or 1 -> 2), and make sure the output is an empty array of the correct width?

Reply:
I will change to using nin!=nout, but the output will be an simply empty in either case. The problem is that I know of no way to call the pybind11 code using a 2-d numpy array that is empty; it seems to be a pybind11 limitation. I hate to drag in boost just for one unit test (though I have tested manually that it works for ndarrays).